lib/optimizely/index.js

```diff
@@ -338,4 +342,19 @@ Optimizely.prototype.__notActivatingExperiment = function(experimentKey, userId)
   return null;
 };
 
+/**
+ * Dispatches an event and executes the designated callback if the dispatch returns a promise
+ * @param  eventToDispatch
+ * @param  callback
+ * @return {null}
+ */
+Optimizely.prototype.__dispatchEvent = function (eventToDispatch, callback) {
+    var eventDispatcherResponse = this.eventDispatcher.dispatchEvent(eventToDispatch, callback);
+    if (typeof eventDispatcherResponse == "object" && !eventDispatcherResponse.hasOwnProperty('uri')) {
```

Comment:
Is the second check still needed: `!eventDispatcherResponse.hasOwnProperty('uri')`?